### PR TITLE
Complete rewrite plus add Zig example

### PR DESCRIPTION
**Pull request by GrayJack**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,38 +1,52 @@
+AR=ar
 CC=clang
-CXX=clang
-LD=ld.lld
```
Comment (cbpudding) [inline Makefile:3]: I don't think the default for LD is LLVM-based

File: Makefile [hunk 2]
```diff
@@ -1,38 +1,52 @@
...
+
+target/debug/libhello_zig.a: src/lib.zig
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@zig build-lib $^ --output-dir target/debug --name hello_zig $(ZIGFLAGS)
+	@printf "\e[92mBuilt $@\e[0m\n"
+
+target/main.o: src/main.c | target
+	@printf "\x1b[0;36mBuilding $@ ...\e[0m\n"
+	@$(CC) -o $@ -c $<
+	@printf "\e[92mBuilt $@\e[0m\n"
```
Comment (cbpudding) [inline Makefile]: I don't like hiding the commands in favor of a bunch of printf statements

File: src/lib.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #include <stdio.h>
 
 void hello_c() {
-	puts("Hello from C!");
+    puts("Hello from C!");
```
Comment (cbpudding) [inline src/lib.c]: Tabs are variable width and can be set to anything while spaces are fixed width. I'm not sure if this change was intentional, but I don't like it.

File: src/lib.cpp [hunk 4]
```diff
@@ -3,5 +3,5 @@
 using namespace std;
 
 extern "C" void hello_cpp() {
-	cout << "Hello from C++!" << endl;
+    cout << "Hello from C++!" << endl;
```
Comment (cbpudding) [inline src/lib.cpp]: Again, spaces instead of tabs.

File: src/main.c [hunk 5]
```diff
@@ -1,12 +1,12 @@
-#include <functions.h>
+#include "functions.h"
```
Comment (cbpudding) [inline src/main.c:1]: I don't really get this change but okay

File: src/main.c [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 
 int main() {
-	for(int i = 0; i < NUMBER_OF_LANGUAGES; i++) {
-		(*hello[i])();
-	}
-	return 0;
+    for(int i = 0; i < NUMBER_OF_LANGUAGES; i++) {
+        (*hello[i])();
+    }
+    return 0;
```
Comment (cbpudding) [inline src/main.c]: Same deal, spaces instead of tabs(does your editor enforce this?)

**Review by cbpudding — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (GrayJack) [reply Makefile]: Should I just show or show + rm printfs?

**Review by cbpudding — APPROVED** [review]
Thanks, it's much better now!

**Pull request merged**